src/services/blocks/BlocksService.spec.ts

```diff
@@ -51,6 +52,12 @@ interface ResponseObj {
  */
 const blocksService = new BlocksService(mockApi);
 
+/**
+ * Object that would be a pointer to AbstractController where the cache would
+ * be stored
+ */
+const that = ({ cache: {} } as unknown) as That;
```

Comment:
Can you ELI5 what a `That` is?

Reply:
Good question, in Javascript there is a convention that is referred to as `this = that`. Originally, the getter and setter for `this.cache` was in `AbstractController`, which `BlocksController` had access too but `BlocksService` didnt. To put it in simple terms when binding or inheriting `this` in one object to another we referr it to `that` 

1. because it cant be called `this`, because `this` already exists
2. That's how the community decided to name it many many years ago :)

But that being said, in this case I agree that the naming is bad and can be way more specific and something like `BlocksControllerThis` could probably be more useful and also get the point across. 

So to answer this question directly as well, `That` here is just the type for the `this` that was getting passed into `BlocksService` from `BlocksController`.

Reply:
But to furthermore I actually completely redesigned this, and that is now completely removed. Things are now tightly scoped, and things are way easier to test now as well.